csv-validator-core/src/main/scala/uk/gov/nationalarchives/csv/validator/MetaDataValidator.scala

```diff
@@ -254,7 +257,15 @@ trait MetaDataValidator {
   }
 
   protected def withReader[B](textFile: TextFile)(fn: JReader => B): B = {
-    managed(new JInputStreamReader(new JFileInputStream(textFile.file.path), textFile.encoding)).map {
+    val inputStream = new JFileInputStream(textFile.file.path);
```

Comment:
I think this looks good, but I would rather see a safer style. Could you adapt this so that everything is managed please, e.g. look at `managed(...).flatMap` or `managed(...).and`

Reply:
Should be improved now. I am not totally familiar with scala-arm, and their documentation on flatMap is a bit lacking. I could have used `map` for all the resources, but I am not sure that would release all the resources once done. `flatMap` with `managed` ought to, though.